src/components/common/countdown-button/countdown-button.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import type { ButtonProps } from 'react-bootstrap'
+import { Button } from 'react-bootstrap'
+
+export interface CountdownButtonProps extends ButtonProps {
+  countdownSeconds: number
+}
+
+/**
+ * Button that starts a countdown on render and is only clickable after the countdown has finished.
+ */
+export const CountdownButton: React.FC<CountdownButtonProps> = ({ countdownSeconds, children, ...props }) => {
+  const intervalRef = useRef<NodeJS.Timeout>()
+  const [secondsRemaining, setSecondsRemaining] = useState(countdownSeconds)
+
+  const onTimerTick = useCallback(() => {
+    setSecondsRemaining((previous) => previous - 1)
```

Comment:
We could deactivate the interval when this has finished so the counter is zero. It should directly work with the return call of the useEffect blow afaik

Reply:
Fixed by using useInterval as suggested by @mrdrogdrog